### PR TITLE
propogate errors

### PR DESCRIPTION
**Pull request by yschimke**
Two changes 

1) TCP Duplex connection treats a EOF or write error as an EOF.  
2) ConsumerMix propogates a CONNECTION_ERROR via onError instead of OnComplete

### CODE REVIEW

File: src/mixins/ConsumerMixin.h [hunk 1]
```diff
@@ -48,8 +48,14 @@ class ConsumerMixin : public Base {
  protected:
   /// @{
   void endStream(StreamCompletionSignal signal) {
-    // FIXME: switch on signal and verify that callsites propagate stream errors
-    consumingSubscriber_.onComplete();
+    switch (signal) {
+      case  StreamCompletionSignal::CONNECTION_ERROR:
```
Comment (stupaq-fb) [inline src/mixins/ConsumerMixin.h]: Can you reformat?

File: src/tcp/TcpDuplexConnection.cpp [hunk 2]
```diff
@@ -64,7 +66,7 @@ void TcpDuplexConnection::readDataAvailable(size_t len) noexcept {
 }
 
 void TcpDuplexConnection::readEOF() noexcept {
-  inputSubscriber_.onComplete();
+  inputSubscriber_.onError(std::runtime_error("connection closed"));
```
Comment (stupaq-fb) [inline src/tcp/TcpDuplexConnection.cpp]: Is it possible to have write err and then read EOF? In this case this will SEGV, right?

File: test/tcp/TcpServer.cpp [hunk 3]
```diff
@@ -72,16 +72,16 @@ class ServerRequestHandler : public DefaultRequestHandler {
 
   void handleFireAndForgetRequest(Payload request) override {
     LOG(INFO) << "ServerRequestHandler.handleFireAndForgetRequest " << request
-              << "\n";
+              << std::endl;
```
Comment (stupaq-fb) [inline test/tcp/TcpServer.cpp]: Why is this needed?

File: test/tcp/TcpServer.cpp [hunk 4]
```diff
@@ -72,16 +72,16 @@ class ServerRequestHandler : public DefaultRequestHandler {
...
   }
 
   void handleMetadataPush(std::unique_ptr<folly::IOBuf> request) override {
     LOG(INFO) << "ServerRequestHandler.handleMetadataPush "
-              << request->moveToFbString() << "\n";
+              << request->moveToFbString() << std::endl;
```
Comment (stupaq-fb) [inline test/tcp/TcpServer.cpp]: Ditto

File: test/tcp/TcpServer.cpp [hunk 5]
```diff
@@ -72,16 +72,16 @@ class ServerRequestHandler : public DefaultRequestHandler {
...
   }
 
   void handleSetupPayload(ConnectionSetupPayload request) override {
-    LOG(INFO) << "ServerRequestHandler.handleSetupPayload " << request << "\n";
+    LOG(INFO) << "ServerRequestHandler.handleSetupPayload " << request << std::endl;
```
Comment (stupaq-fb) [inline test/tcp/TcpServer.cpp]: Ditto.

File: src/tcp/TcpDuplexConnection.cpp (hunk 2 above)
Comment (yschimke) [inline src/tcp/TcpDuplexConnection.cpp]: checking inputSubscriber_ is enough, right?

Comment (stupaq-fb) [inline src/tcp/TcpDuplexConnection.cpp]: I believe so.

File: test/tcp/TcpServer.cpp (hunk 3 above)
Comment (lehecka) [inline test/tcp/TcpServer.cpp]: we shouldn't be writing new lines into LOG statements, it would make grep filtering impossible.

Comment (yschimke) [inline test/tcp/TcpServer.cpp]: That is gone already, just showing because of github reviews AFAIK.  I will merge this is it passes tests.
